agent/engine/dependencygraph/graph.go

```diff
@@ -211,8 +218,13 @@ func executionCredentialsResolved(target *apicontainer.Container, id string, man
 		return true
 	}
 
-	_, ok := manager.GetTaskCredentials(id)
-	return ok
+	if target.ExecutionCredentialsID != "" {
+		_, ok := manager.GetContainerCredentials(id)
+		return ok
```

Comment:
I wonder if we want to fall back to task credentials when `!ok` here?

Reply:
I don't think so --- when a container has container credentials specified, it could have different set of privilege with task credentials, fall back to task credentials might cause unexpected access to resources

Reply:
could we double check that that's the desired behavior? 
Previous to this change, containers get credentials from tasks level, fall back to instance-level creds. Why should this be different?

Reply:
There are two fallback cases:
1. Fallback if there are no container credentials. We DEFINITELY want to fall back in this case- its part of the defined CX. If there is no container role, then we must use the task def role. 
2. Fallback if we fail to use the credentials somehow (ex fail to pull secret). I am unsure about this case. Having a chat conversation folks on this.

Reply:
For the container get credentials from task level and fall back to instance-level, can you point where we did it? we do have back up for credentials providers but I don't think there is a fall back strategy for credentials. Just checked with Wesley, his opinion is that if we don't have existing fall back strategies then we might not want to introduce a new one.

Reply:
Sorry for the confusion. You are correct in that only credential providers which are related to task roles have the fallback mechanism. Task execution role doesn't.